### PR TITLE
Extend i.MX6 LVDS support

### PR DESCRIPTION
**Pull request by Deryugin**
* Refactor IPU
* Add 24-bit mode for i.MX6 LVDS
* Extent IPU support
* Minor fixes for `fb_direct_access`
* i.MX6 GPIO fix
* Create template for VAR SOM MX6 board
* Other minor improvements (refactor etc)

### CODE REVIEW

File: src/drivers/gpu/ipu_v3/ipu_param_mem.h [hunk 1]
```diff
@@ -13,16 +13,19 @@
 #ifndef __INCLUDE_IPU_PARAM_MEM_H__
 #define __INCLUDE_IPU_PARAM_MEM_H__
 
+#include <errno.h>
+#include <stdint.h>
+
 #include "uboot_ipu_compat.h"
 #include "ipu_priv.h"
-#include <errno.h>
+
```
Comment (Deryugin) [inline src/drivers/gpu/ipu_v3/ipu_param_mem.h]: Лишняя пустая строка.

File: src/drivers/gpu/ipu_v3/ipu_priv.h [hunk 2]
```diff
@@ -12,13 +12,16 @@
...
+#include <stdint.h>
+
 #include <kernel/irq.h>
 #include "uboot_ipu_compat.h"
 #include <util/log.h>
 /* Globals */
 extern int dmfc_type_setup;
 
-#define IDMA_CHAN_INVALID	0xFF
+#define IDMA_CHAN_INVALID	    0xFF
```
Comment (Deryugin) [inline src/drivers/gpu/ipu_v3/ipu_priv.h]: Unneccessary space.

File: src/drivers/gpu/ipu_v3/uboot_ipu_compat.h [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
+//typedef uint32_t resource_size_t;
+typedef uint32_t dma_addr_t;
+
 
 #define __raw_readl(addr) REG32_LOAD(addr)
 #define readl(addr) REG32_LOAD(addr)
 #define __raw_writel(value, addr) REG32_STORE(addr, value)
 #define writel(value, addr) REG32_STORE(addr, value)
 
+#if 0
```
Comment (Deryugin) [inline src/drivers/gpu/ipu_v3/uboot_ipu_compat.h]: It seems that this can be safely removed

File: src/drivers/gpu/ipu_v3/uboot_ipu_compat.h [hunk 4]
```diff
@@ -55,15 +59,15 @@ typedef uint32_t resource_size_t;
...
 
-typedef uint32_t dma_addr_t;
-typedef uint8_t u8;
-typedef uint16_t u16;
-typedef uint32_t u32;
-
+//typedef uint8_t u8;
+//typedef uint16_t u16;
+//typedef uint32_t u32;
+#if 0
```
Comment (Deryugin) [inline src/drivers/gpu/ipu_v3/uboot_ipu_compat.h]: It seems that this can be safely removed

File: src/drivers/serial/imx_uart/imx_uart.c [hunk 5]
```diff
@@ -17,7 +17,7 @@ EMBOX_UNIT_INIT(uart_init);
 
 #define IOMUXC_BASE	OPTION_GET(NUMBER,iomuxc_base)
 #define UART_NUM	OPTION_GET(NUMBER,num)
-#define IRQ_NUM		(58 + UART_NUM)
+#define IRQ_NUM		((32 + 26) + UART_NUM)
```
Comment (Deryugin) [inline src/drivers/serial/imx_uart/imx_uart.c]: Why this is neccesary?

File: src/drivers/net/fec/fec.c [hunk 6]
```diff
@@ -294,13 +279,7 @@ static void _reset(struct net_device *dev) {
...
 
 	fec_tbd_init(dev->priv);
-#if 0
-#if FEC_SPEED == 1000
-	REG32_STORE(ENET_ECR, 0xF0000100 | ENET_SPEED);
-#else
-	REG32_STORE(ENET_ECR, 0xF0000100);
-#endif
-#endif
+
```
Comment (Deryugin) [inline src/drivers/net/fec/fec.c]: Remove this empty line, please.

File: src/drivers/serial/imx_uart/imx_uart.c (hunk 5 above)
Comment (anton-bondarev) [reply src/drivers/serial/imx_uart/imx_uart.c]: for greater clarity. from an documentation irq number 26+uart_num

File: src/drivers/net/fec/fec.c (hunk 6 above)
Comment (anton-bondarev) [reply src/drivers/net/fec/fec.c]: I think it's better with one

File: src/drivers/serial/imx_uart/imx_uart.c (hunk 5 above)
Comment (Deryugin) [reply src/drivers/serial/imx_uart/imx_uart.c]: It's not clear what are `32` and `26`, so it's better to declare it as a macro.

**Comment by Deryugin** [comment]
I think we should use `netmanager` in new templates, this way we make sure it works (and `start_script.inc` looks the way better).

**Pull request merged**